no_agent branch broken by a test sentence

In the no_agent branch, the following sentence throws a "scala.NotImplementedError":

"Green, do you see any rubble here? "

I assume this has something to do you with dependencies traversing the new agent attachments. I will paste the full error code below:

`play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[RuntimeException: scala.NotImplementedError: an implementation is missing]]
        at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:255)
        at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:182)
        at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:251)
        at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:250)
        at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:414)
        at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
        at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:70)
        at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:68)
        at scala.concurrent.impl.Promise$KeptPromise$Kept.onComplete(Promise.scala:368)
Caused by: java.lang.RuntimeException: scala.NotImplementedError: an implementation is missing
        at play.api.mvc.ActionBuilder$$anon$2.apply(Action.scala:422)
        at play.api.mvc.Action.$anonfun$apply$2(Action.scala:96)
        at play.api.libs.streams.StrictAccumulator.$anonfun$mapFuture$4(Accumulator.scala:174)
        at scala.util.Try$.apply(Try.scala:209)
        at play.api.libs.streams.StrictAccumulator.$anonfun$mapFuture$3(Accumulator.scala:174)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:52)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:52)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:52)
        at play.api.libs.streams.StrictAccumulator.run(Accumulator.scala:207)
        at play.core.server.AkkaHttpServer.executeAction(AkkaHttpServer.scala:297)
Caused by: scala.NotImplementedError: an implementation is missing
        at scala.Predef$.$qmark$qmark$qmark(Predef.scala:284)
        at org.clulab.asist.TomcatActions.mkAgent(TomcatActions.scala:80)
        at org.clulab.asist.TomcatActions.$anonfun$convertAgent$3(TomcatActions.scala:38)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
        at scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:74)
        at scala.collection.TraversableLike.map(TraversableLike.scala:234)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at org.clulab.asist.TomcatActions.convertAgent(TomcatActions.scala:38)
        at org.clulab.asist.TomcatActions.$anonfun$convertAgents$1(TomcatActions.scala:32)
`

--- a/src/main/resources/org/clulab/asist/grammars/ent_template.yml
+++ b/src/main/resources/org/clulab/asist/grammars/ent_template.yml
@@ -2,7 +2,7 @@ vars: org/clulab/asist/grammars/vars.yml
 
 rules:
   - name: ${ name }_detection
-    priority: ${ priority }+
+    priority: ${ priority }
     label: ${ label }
     type: token
     keep: true

--- a/src/main/resources/org/clulab/asist/grammars/entities.yml
+++ b/src/main/resources/org/clulab/asist/grammars/entities.yml
@@ -75,7 +75,7 @@ rules:
     type: token
     keep: true
     pattern: |
-      [lemma=/(?i)^(${ you_triggers })$/ & tag=/PRP/]
+      [word=/(?i)^(${ you_triggers })$/ & tag=/PRP/]
 
   - name: we_token_capture
     label: Team
@@ -149,4 +149,29 @@ rules:
     keep: false
     pattern: |
       others | other (?! [tag=/^N|CD/]) | (?<= other) [tag=/CD|^N/]
+
+  - name: possesive_pronoun_my
+    priority: ${ rulepriority }
+    label: Self
+    type: token
+    keep: false
+    pattern: |
+      [tag=/PRP/ & word=/my/]
+
+  - name: possesive_pronoun_you
+    priority: ${ rulepriority }
+    label: You
+    type: token
+    keep: false
+    pattern: |
+      [tag=/PRP/ & word=/your/]
+
+  - name: possesive_pronoun_other
+    priority: ${ rulepriority }
+    label: Other
+    type: token
+    keep: false
+    pattern: |
+      [tag=/PRP/ & word=/his|her|their/]
+# the above regex means: match anything but "m" and "o", which will exclude 1st and 2nd person possessive prns
 

--- a/src/main/resources/org/clulab/asist/grammars/locations.yml
+++ b/src/main/resources/org/clulab/asist/grammars/locations.yml
@@ -18,7 +18,24 @@ rules:
     keep: true
     pattern: |
       /${room_singleword_surface_regex}/ |
-      /${room_twoword_first_word_regex}/ room
+      /(?i)${room_twoword_first_word_regex}/ [tag=/POS/]? /(?i)${room_twoword_second_word_regex}/
+
+  - name: storage_room_numb
+    priority: ${ first_priority }
+    label: Infrastructure
+    type: token
+    keep: true
+    pattern: |
+      [word=/(?i)storage/] [word=/(?i)room/]? /(?i)^[A-Z]/
+
+
+  - name: room_numbered
+    priority: ${ first_priority }
+    label: Infrastructure
+    keep: true
+    pattern: |
+      trigger = [lemma=/(?i)^room/]
+      number: Number = >/nummod/
 
   - name: north
     priority: ${ first_priority }

--- a/src/main/resources/org/clulab/asist/grammars/master.yml
+++ b/src/main/resources/org/clulab/asist/grammars/master.yml
@@ -8,12 +8,12 @@ rules:
   # Extract the entities
   - import: org/clulab/asist/grammars/entities.yml
     vars:
-      rulepriority: "1"
+      rulepriority: "1+"
 
   # Extract the items
   - import: org/clulab/asist/grammars/items.yml
     vars:
-      rulepriority: "1"
+      rulepriority: "1+"
 
   # Extract the sentiments
   - import: org/clulab/asist/grammars/sentiments.yml

--- a/src/main/resources/org/clulab/asist/grammars/questions.yml
+++ b/src/main/resources/org/clulab/asist/grammars/questions.yml
@@ -24,14 +24,24 @@ rules:
       location: Location? = >/${positional_preps}/
 
   - name: information_gathering_question_that
-    example: "What is that?  What is that over there"
+    example: "What is that? What is that in the library?"
     priority: ${ rulepriority }
     label: Question
     action: removeResearcher
     pattern: |
-      trigger = @QuestionParticle
+      trigger = @QuestionParticle (?= [lemma=be])
       topic: Concept = <nsubj [lemma=/be|do/] >dep
       location: Location? = <nsubj [lemma=/be|do/] >dep >/${positional_preps}/
+
+  - name: information_gathering_question_that2
+    example: "What's that over there?"
+    priority: ${ rulepriority }
+    label: Question
+    action: removeResearcher
+    pattern: |
+      trigger = @QuestionParticle (?= [lemma=be])
+      topic: Concept = >nsubj
+      location: Location? = >/${positional_preps}/
 
   - name: information_gathering_question_clarification
     example: "What is the plan?"
@@ -41,7 +51,7 @@ rules:
     pattern: |
       trigger = @QuestionParticle (?= [lemma=be])
       topic: Concept = >nsubj
-      location: Location? = >/${positional_preps}/
+      location: Location? = >nsubj >/${positional_preps}/
 
 
 #leave this here so I can remember how to do these complex dependencies

--- a/src/main/resources/org/clulab/asist/grammars/taxonomy.yml
+++ b/src/main/resources/org/clulab/asist/grammars/taxonomy.yml
@@ -52,6 +52,7 @@
           - Disagreement
         - Clear
         - RoleSwitch
+        - RoleDeclare
         - ReportLocation
         - KnowledgeSharing
       - ComplexActions:

--- a/src/main/resources/org/clulab/asist/grammars/team_communication.yml
+++ b/src/main/resources/org/clulab/asist/grammars/team_communication.yml
@@ -61,7 +61,17 @@ rules:
     pattern: |
       trigger = [lemma=/(?i)^(${ role_switch_triggers })/ & tag=/^V/]
       agent: Entity? = >/(${agents})/
-      target: Role? = >/nmod_to|xcomp/
+      target: Role = >/nmod_to|xcomp/
+
+  - name: role_declare
+    priority: ${ rulepriority }
+    label: RoleDeclare
+    example: "I am an engineer"
+    action: removeResearcher
+    pattern: |
+      trigger = [lemma=/(?i)^be/]
+      target: Role = <cop
+      agent: Entity = <cop >/(${agents})/
 
   - name: role_switch_change_to
     priority: ${ rulepriority }

--- a/src/main/resources/org/clulab/asist/grammars/time.yml
+++ b/src/main/resources/org/clulab/asist/grammars/time.yml
@@ -15,3 +15,11 @@ rules:
     type: token
     pattern: |
       (?<value> [word=/(?i)last|first/ | tag=/^DT/] [word=/(?i)few|several|couple/]) of? (?<trigger> [lemma=/(?i)second|minute|hour|day|year/])
+
+  - name: time_mention
+    label: Time
+    priority: 1
+    type: token
+    keep: false
+    pattern: |
+      [lemma=/time/]

--- a/src/main/resources/org/clulab/asist/grammars/triggers.yml
+++ b/src/main/resources/org/clulab/asist/grammars/triggers.yml
@@ -20,8 +20,11 @@ fire_triggers: "fire|blaze|flame"
 
 infrastructure_triggers: "room|door|building|stairs|window|floor|ceiling|house|roof|bathroom|hall|entrance|library|balcony"
 
-room_singleword_surface_regex: "armory|attic|ballroom|basement|bathroom|bedroom|boardroom|cabin|cellar|classroom|courtroom|cubby|den|entry|garage|gym|hall|hallway|kitchen|kitchenette|library|lobby|loft|lounge|lunchroom|mailroom|office|restroom|schoolroom|stockroom|storeroom|studio|study|suite|washroom|workroom"
-room_twoword_first_word_regex: "back|boiler|conference|control|dining|engine|front|ladies'|laundry|living|locker|men's|utility|waiting|weight|women's"
+room_singleword_surface_regex: "(cafe|buffet|chophouse|armory|attic|ballroom|basement|bathroom|bedroom|boardroom|cabin|cellar|classroom|courtroom|cubby|den|entry|garage|gym|hall|hallway|kitchen|kitchenette|library|lobby|loft|lounge|lunchroom|mailroom|office|restroom|schoolroom|stockroom|storeroom|studio|study|suite|washroom|workroom)"
+
+room_twoword_first_word_regex: "(cubicle|sundollar|Herbalife|limping lamb|lamb|storage|King Chris|king|Chris|back|boiler|conference|control|dining|engine|front|ladies|laundry|living|locker|men|utility|waiting|weight|women|break)"
+
+room_twoword_second_word_regex: "(room|cafe|office|terrace|area|storage|coffee|chophouse)"
 
 switch_triggers: "switch|toggle|lever|door"
 

--- a/src/main/scala/org/clulab/asist/TomcatActions.scala
+++ b/src/main/scala/org/clulab/asist/TomcatActions.scala
@@ -66,22 +66,7 @@ class TomcatActions(
     Interval(start, end)
   }
 
-  def mkAgent(m: Mention): Attachment = {
-    // from taxonomy, we have...
-    //    - Other
-    //    - Self
-    //    - You
-    //    - Team
-    val agentType = m.label match {
-      case "Other" => Agent.OTHER
-      case "Self" => Agent.SELF
-      case "You" => Agent.YOU
-      case "Team" => Agent.TEAM
-      case _ => ??? // fixme?
-    }
-    Agent(m.text, agentType)
-  }
-
+  def mkAgent(m: Mention): Attachment = Agent(m.text, m.label)
 
 /** Keeps the longest mention for each group of overlapping mentions **/
   def keepLongest(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {

